### ISSUE TITLE
EC2 natting issues

### ISSUE THREAD

**Issue by debitcash** [friedrich/hans#27]
Hey guys, does anybody have a good guide how to use it?
Ive tried numerous guides and none of them work, using pingable EC2 and kali.
Im pretty sure I screwed up with NAtting somewhere, but official docs do not mention anything about it
Thanks

**Comment by norohind** [comment]
Hello. In my limited experience with EC2 it had a publicly accessible address, but the interface in the instance had a private address. So there might be some NAT in front of EC2, which works fine and maps all udp and tcp ports 1:1 from the ip address to the instance, but refuses to work with icmp. I'd check with tcpdump/tshark if you receive any icmp packets from the client.

**Comment by debitcash** [comment]
So after running tcpdump, I got the following, which shows that I am getting the pings and they are visible on the ec2( 

13:52:57.193742 (my ISP's ip) > (EC2's internal IP) ICMP echo request, id 32510, seq 39633, length 73
13:52:57.194916 (my ISP's ip) > (EC2's internal IP) ICMP echo request, id 32510, seq 39633, length 73
13:52:58.179839 (my ISP's ip) > (EC2's internal IP) ICMP echo request, id 32510, seq 39633, length 13
13:53:00.183859 (my ISP's ip)> (EC2's internal IP) ICMP echo request, id 32510, seq 39633, length 13

The IDs, sequence numbers, and lengths match what I can see in outgoing ICMP packets from my laptop's (client) port that is connected to the internet. 
I've discovered tho, that client's tun0 port sends two SYN packets to the website I'm trying to reach, followed by a series of TCP retransmission packets no ACK no nothing just a spam of retransmissions. Additionally, my client does not receive ICMP responses on its internet-connected port.

So overall what I figured- request does go to the EC2, but nothing is returned.

Any ideas what should I do next?
(pls bear with me lol)
